packages/server-node/src/launch/glsp-server-launcher.ts

```diff
@@ -41,7 +41,7 @@ export abstract class GLSPServerLauncher<T = undefined> {
     start(startParams: T): MaybePromise<void> {
         if (!this.running) {
             const result = this.run(startParams);
-            this.running = false;
+            this.running = true;
```

Comment:
We should set running to `true` before we call `this.run`.
Otherwise calling `shutdown()`  will fail in non-async run implementations;

Reply:
Good catch! updated with next commit.